sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformationExec.scala

```diff
@@ -161,6 +161,7 @@ case class ScriptTransformationExec(
 
               if (scriptOutputReader != null) {
                 if (scriptOutputReader.next(scriptOutputWritable) <= 0) {
```

Comment:
Does `<=0` always implies a terminated or terminating process?

Reply:
No, I find a way to reproduce. Just run this code in laptop for some times, it will show that when `inputStream` return -1, but the `proc` is alive. Actually, `ScriptTransformationExec ` also uses `inputStream` to check the end of stream.

``` scala 
import scala.collection.JavaConverters._

object SparkTEST {
  def main(args: Array[String]): Unit = {
    var i = 0
    for (i <- 0 until (50)) {
      check_script()
    }
  }

  def check_script() = {
    val script = "python /Users/buytedance/code/github/error_python.py"
    val cmd = List("/bin/bash", "-c", script)
    val builder = new ProcessBuilder(cmd.asJava)
    val proc = builder.start()
    val inputStream = proc.getInputStream
    var msg = ""
    if (inputStream.read() <= 0) {
      msg = s"Cannot read inputStream and proc status is: ${if (proc.isAlive) "alive" else "dead"}"
    }
    println(msg)
  }
}
```

Reply:
Oh...I mean, is it possible that `scriptOutputReader.next` returns `<=0` but the process hang?
If it's possible, we may need to destroy the process before `proc.waitFor()`.

Reply:
The process not hangs, but just in a terminating state and alive. It is really hard to reproduce in local mode, but it happened some times in our online cluster.